packages/@glimmer/runtime/lib/dom/sanitized-values.ts

```diff
@@ -30,8 +29,30 @@ export function requiresSanitization(tagName: string, attribute: string): boolea
   return checkURI(tagName, attribute) || checkDataURI(tagName, attribute);
 }
 
+let protocolForUrl: (url: string) => string;
+
+if (typeof document !== 'undefined') {
+  let parsingNode = document.createElement('a');
+
+  protocolForUrl = (url: string) => {
+    parsingNode.href = url;
+    return parsingNode.protocol;
+  };
+} else {
+  let nodeURL = module.require('url');
+
+  protocolForUrl = (url: string) => {
+    let protocol = null;
+
+    if (typeof url === 'string') {
+      protocol = nodeURL.parse(url).protocol;
+    }
+
+    return protocol === null ? ':' : protocol;
+  };
+}
```

Comment:
I would prefer something like:

```suggestion
if (typeof URL === 'function') {
  // supports Node >= 10 and all browsers except IE11
  protocolForUrl = (urlString: string) => {
    let url = new URL(urlString, typeof document !== 'undefined' ? document.baseURI : undefined);
    
    return url.protocol;
  };
} else if (typeof document !== 'undefined') {
  // supports IE11 environment
  let parsingNode = document.createElement('a');

  protocolForUrl = (url: string) => {
    parsingNode.href = url;
    return parsingNode.protocol;
  };
}
```

This unifies implementation for all modern platforms, and makes it really easy to know what to remove when IE11 support is no longer needed.

Also, see [this note in the Node URL module documentation](https://nodejs.org/api/url.html#url_url_parse_urlstring_parsequerystring_slashesdenotehost):

> Use of the legacy url.parse() method is discouraged. Users should use the WHATWG URL API. Because the url.parse() method uses a lenient, non-standard algorithm for parsing URL strings, security issues can be introduced. Specifically, issues with host name spoofing and incorrect handling of usernames and passwords have been identified.

Reply:
So I considered doing this, the main reason I didn't is I was afraid about matching behavior. For instance, `url.parse('image.jpg');` works as a relative url and does not error, but `new URL('image.jpg', undefined);` does not.

So in your solution, in Node, that would throw an error. I think we would need to wrap `new URL` in a try/catch, because if a _valid_ URL is passed it'll parse correctly and everything will be fine, but if a _relative_ URL is passed it'll throw.